ui/raidboss/data/07-dt/raid/r8s.ts

```diff
@@ -1,14 +1,1907 @@
+  // Also used for Elemental Purge in Phase 2
+  'predation': '0017',
+  // Stony tether from Wolf of Stone
+  'stoneTether': '014F',
+  // Windy Tether from Wolf of Wind
+  'windTether': '0150',
+  // Big, pulsing, 4-arrow stack marker
+  'eightHitStack': '013C',
+  // Spread marker used in Terrestial Rage and Beckon Moonlight
+  'spread': '008B',
+  // Stack marker used in Terrestial Rage and Beckon Moonlight
+  'stack': '005D',
+  // Blue circle marker with spikes used for Ultraviolent Ray target in Phase 2
+  'ultraviolent': '000E',
+  // Passable tether used for Twofold Tempest in Phase 2
+  'twofoldTether': '0054',
+  // Blue arrows going counterclock on the boss
+  'counterclockwise': '01F6',
+  // Orange arrows going clockwise on the boss
+  'clockwise': '01F5',
+  // Blue far tether in Lone Wolf's Lament in Phase 2
+  'farTether': '013E',
+  // Green close tether in Lone Wolf's Lament in Phase 2
+  'closeTether': '013D',
+} as const;
+
+const stoneWindOutputStrings = {
+  stoneWindNum: {
+    en: '${debuff} ${num}',
+  },
+  stone: {
+    en: 'Stone',
+  },
+  wind: {
+    en: 'Wind',
+  },
+  unknown: Outputs.unknown,
+};
+
+const moonlightOutputStrings = {
+  ...Directions.outputStrings8Dir,
+  safeQuad: {
+    en: '${quad}',
+  },
+  safeQuadrants: {
+    en: '${quad1} => ${quad2}',
+  },
+};
+
+const championOutputStrings = {
+  clockwise: Outputs.clockwise,
+  counterclockwise: Outputs.counterclockwise,
+  in: Outputs.in,
+  out: Outputs.out,
+  donut: {
+    en: 'Donut',
+  },
+  sides: Outputs.sides,
+  mechanics: {
+    en: '(${dir}) ${mech1} => ${mech2} => ${mech3} => ${mech4} => ${mech5}',
+  },
+  left: Outputs.left,
+  right: Outputs.right,
+  leftSide: {
+    en: 'Left Side',
+  },
+  rightSide: {
+    en: 'Right Side',
+  },
+  unknownSide: {
+    en: '??? Side',
+  },
+  dirMechanic: {
+    en: '${dir} ${mech}',
+  },
+};
+
+// Platform, Mechs
+// S = 0, SW = 1, NW = 2, NE = 3, SE = 4
+const championClockOrders: ChampionOrders = {
+  0: ['donut', 'in', 'out', 'in', 'sides'],
+  1: ['sides', 'donut', 'in', 'out', 'in'],
+  2: ['in', 'sides', 'donut', 'in', 'out'],
+  3: ['out', 'in', 'sides', 'donut', 'in'],
+  4: ['in', 'out', 'in', 'sides', 'donut'],
+};
+const championCounterOrders: ChampionOrders = {
+  0: ['donut', 'sides', 'in', 'out', 'in'],
+  1: ['in', 'donut', 'sides', 'in', 'out'],
+  2: ['out', 'in', 'donut', 'sides', 'in'],
+  3: ['in', 'out', 'in', 'donut', 'sides'],
+  4: ['sides', 'in', 'out', 'in', 'donut'],
+};
+
+// Return the combatant's platform by number
+const getPlatformNum = (
+  x: number,
+  y: number,
+): number => {
+  // S Platform
+  if (x > 90 && x < 108 && y > 107)
+    return 0;
+
+  // SW Platform
+  if (x <= 92 && y > 95)
+    return 1;
+
+  // NW Platform
+  if (x <= 100 && y <= 95)
+    return 2;
+
+  // NE Platform
+  if (x > 100 && y <= 95)
+    return 3;
+
+  // SE Platform
+  if (x >= 109 && y > 95)
+    return 4;
+
+  return -1;
+};
```

Comment:
There's a lot of comparisons here against position x/y that could potentially be sensitive to floating point rounding problems.  I didn't manually review the data, but any time I see an `x` or `y` comparison against exactly 100, it makes me nervous.

It may be more safe to check `x < 101` and `x > 101` instead?

Reply:
The values are rough estimates based off some fflogs position collection. It certainly could use more precise/algorithmic values than my guesstimates. The interesting thing is if we did get a value that is that close to 100 it likely meant they were transferring between platforms and our data is wrong anyways as we aren't tracking a vector.